lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -15,10 +15,45 @@ defmodule Mix.Compilers.Elixir do
   between modules, which helps it recompile only the modules that
   have changed at runtime.
   """
-  def compile(manifest, srcs, skip, exts, dest, force, on_start) do
+  def compile(manifest_map, srcs, skip, exts, dest, force, on_start) do
     keep = srcs -- skip
-    all  = Mix.Utils.extract_files(keep, exts)
-    {all_entries, skip_entries} = parse_manifest(manifest, keep)
```

Comment:
Lots of lookup maps here.

The `universe` is all files, even those outside of the specified `elixirc_paths`

`all` is all files within the `elixirc_paths`. For manifests, `keep` is used for the manifests we are compiling.

The basic flow is:

1. Get all modified and new files (including files with compile time dependencies that have manifests that were modified)
2. Compile everything
3. Group entries by path and write each manifest separately.